kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/impl/DefaultSharedIndexInformer.java

```diff
@@ -208,9 +209,6 @@ private void handleDeltas(Deque<AbstractMap.SimpleEntry<DeltaFIFO.DeltaType, Obj
 
   @Override
   public void addIndexers(Map<String, Function<T, List<String>>> indexers) {
-    if (started) {
-      throw new IllegalStateException("Cannot add indexers to a running informer.");
-    }
```

Comment:
This is a direct port  of client-go.

In the original implementation. you cannot add indexers to a running informer (https://github.com/kubernetes/client-go/blob/8c8fa70f7a2acc191d4787327639621f69992efd/tools/cache/shared_informer.go#L441)

What would be the effects of removing this condition?

Reply:
The condition has not been removed - it's moved to the Cache.addIndex method as the primary enforcement point.